File: andrew_ward/week_02/warm_ups/nucleotides/nucleotides.js
var nucleotides = function(str) {
  var a = 0;
  var c = 0;
  var g = 0;
  var t = 0;
  var u = 0;
  for (var i = 0; i < str.length; i++) {
    if (str[i] === 'a') {
      a += 1;
    }
  }
  for (var x = 0; x < str.length; x++) {
    if (str[x] === 'c') {
      c += 1;
    }
  }
  for (var y = 0; y < str.length; y++) {
    if (str[y] === 'g') {
      g += 1;
    }
  }
  for (var m = 0; m < str.length; m++) {
    if (str[m] === 't') {
      t += 1;
    }
  }
  for (var n = 0; n < str.length; n++) {
    if (str[n] === 'u') {
      u += 1;
    }
  }
  if (a + c + g + t + u === 0) {
    return 'There are no nucleotides here.';
  }
  console.log('There are ' + a + ' A nucleotides.');
  console.log('There are ' + c + ' C nucleotides.');
  console.log('There are ' + g + ' G nucleotides.');
  console.log('There are ' + t + ' T nucleotides.');
  console.log('There are ' + u + ' U nucleotides.');
  console.log('In total, there are ' + (a + c + g + t + u) + ' nucleotides.');
}

File: andrew_ward/week_04/warmUps/nucleotides/nucleotides.rb
# There are 5 types of nucleotides. 4 of these occur in DNA: A, C, G, and T. 4 occur in RNA: A, C, G, U.

# There are no other nucleotides.

# Make sure that you validate it has nucleotides!

# Find how many of each nucleotides are in a string that you pass in to a function

require "rainbow"

nucleotides = {:A => 0, :C => 0, :G => 0, :T => 0, :U => 0}

def nucleotide_detector string, nucleotides

  string.each_char do |char|
    nucleotides.each do |key,val|
      if char.to_sym == key
        nucleotides[key] += 1
      end
    end
  end

  total_nucleotides = 0

  nucleotides.each do |key,val|
    total_nucleotides += nucleotides[key]
  end

  puts Rainbow("There are #{total_nucleotides} nucleotides in total.").green
  puts Rainbow(nucleotides).cyan
end

print Rainbow("What string would you like to check for nucleotides? ").red
str = gets.chomp.upcase

nucleotide_detector str, nucleotides



